bgpd/bgp_network.c

```diff
@@ -219,14 +219,13 @@ int bgp_md5_unset(struct peer_connection *connection)
 	return bgp_md5_set_password(connection, NULL);
 }
 
-int bgp_set_socket_ttl(struct peer_connection *connection)
+int bgp_set_socket_ttl(struct peer_connection *connection, int sock)
```

Comment:
This is the incorrect fix.  The bgp_set_socket_ttl should be connection based and as such we should be passing in the correct connection here.

Reply:
What would be the right fix here?

Reply:
`connection1->fd = bgp_sock;` like we do for dynamic peers?

Reply:
Fixed.